code/datums/components/footstep.dm

```diff
@@ -86,14 +86,14 @@
 			GLOB.barefootstep[FOOTSTEP_RESIN][3] + e_range)
 		return
 	var/mob/living/carbon/human/H = parent
-
-	if(H.shoes) //are we wearing shoes
-		playsound(T, pick(GLOB.shoefootstep[T.shoefootstep][1]),
-			GLOB.shoefootstep[T.shoefootstep][2] * volume,
+	if (!CHECK_BITFIELD(H.flags_pass,(PASSFIRE)))//are we flying?
```

Comment:
A **passfire** check for **shoes sound** check should be pretty self explanatory as to why it's bad